### ISSUE TITLE
space-in-brackets and template string

### ISSUE THREAD

**Issue by romeovs** [eslint/eslint#2456]
The `space-in-brackets` rule has a number of exceptions to match different this like `propertyName`, etc.

It does not, however, cover the case of brackets in template strings. Eg.
With:
```yaml
rules:
  space-in-brackets:
    - 2
    - always
```

the following snippet has an error:
```js
var x = 10;
var foo = `bar ${x} baz`;
```
because there is no space in the brackets of the interpolation (`${x}`).

I would like to force spaces in brackets, except for in interpolations so I an exception
called `templateInterpolation` or similar to disable the rule in that case.

**Comment by nzakas** [comment]
Can you paste the actual output? And are you using babel-eslint?

**Comment by romeovs** [comment]
hmm, I changed my config quite a lot and cannot reproduce the errors.
Let me try again tomorrow!

**Comment by xjamundx** [comment]
Given `ESLint 0.20.00` with the following file:
```js
var x = 10;
var foo = `bar ${x} baz`;
```
I cannot reproduce using:
` eslint --reset --rule "{strict:0, space-in-brackets: [2,'always']}" --env es6 test.js`

It indicates there are no errors.

**Comment by romeovs** [comment]
I can't reproduce it either. Strange...
I'm closing this. Sorry for the hassle.
